### ISSUE TITLE
Auto-compilation of module changes working dir of vterm

### ISSUE THREAD

**Issue by purcell** [akermu/emacs-libvterm#243]
Loving `vterm`, thanks! I noticed a quirk which affects me each time the version is bumped:

### Steps to reproduce:

* Install a new version of `vterm`
* Run `M-x vterm`

### Expected result

* Module is compiled
* `vterm` session is started with the default directory of the buffer from which it launched

### Actual result

* Module is compiled
* `vterm` session starts with the working directory set to the installation dir of the new package, e.g. `~.emacs.d/elpa/vterm-20200210.704`

**Comment by purcell** [comment]
I should note that I looked at the code and it wasn't immediately obvious why this should happen.

**Comment by Sbozzolo** [comment]
In `vterm-module-make`, we set `default-directory` to the folder where `vterm.el` was downloaded. I don't fully understand what is going with the binding of this variable, but it seems that this is why you see the reported behavior. 

I am working on an improved version of `vterm-module-make` that allows the user to choose which libvterm to use (the system one, or the one downloaded from git). This is in PR #240. In the latest commit, I removed all the references to `default-directory`, and added a `cd` to the compilation command. As a result, my tests indicate that `vterm` now behaves as expected.

The commit history in #240 is messy (most commits are buggy due to a missing blank space - I'll clean it up before I merge it), but the last commit should be fine regardless the version of libvterm you want to use.
